packages/button/src/button.interface.ts

```diff
@@ -0,0 +1,57 @@
+export interface BaseButtonProps {
+  // default html button props
+  className?: string;
+  children?: React.ReactNode;
+
+  /**
+   * a props to set button variant type
+   * @default solid
+   */
+  variants?: 'solid' | 'outline' | 'ghost' | 'link';
+  /**
+   * a props to set button color & background color
+   * @default gray
+   */
+  colorScheme?: any;
```

Comment:
ini kenapa dibuat as any va?, kayaknya propsnya juga ga perlu di forward ke komponen style nya since di komponent style nya ga punya colorscheme, cmiiw ya.

Reply:
oh shoot, I think, I forgot to change it back to string since the last time I faced a warning from typescript that, 
`Type 'string' is not assignable to type 'undefined'` I'll change it back to string later.

and for the **colorScheme**, I still need to pass it to the **useResolvedThemes**, because I still use it through the styled function props. As you can see on each styled function variant, I used a color Scheme through the styled function props.